src/SFML/Graphics/RenderTextureImplFBO.cpp

```diff
@@ -84,17 +86,17 @@ namespace
         sf::Uint64 contextId = sf::Context::getActiveContextId();
 
         // Destroy active frame buffer objects
-        for (std::set<std::map<sf::Uint64, unsigned int>*>::iterator frameBuffersIter = frameBuffers.begin(); frameBuffersIter != frameBuffers.end(); ++frameBuffersIter)
+        for (std::unordered_set<std::unordered_map<sf::Uint64, unsigned int>*>::iterator frameBuffersIter = frameBuffers.begin(); frameBuffersIter != frameBuffers.end(); ++frameBuffersIter)
         {
-            for (std::map<sf::Uint64, unsigned int>::iterator iter = (*frameBuffersIter)->begin(); iter != (*frameBuffersIter)->end(); ++iter)
+            for (std::unordered_map<sf::Uint64, unsigned int>::iterator iter = (*frameBuffersIter)->begin(); iter != (*frameBuffersIter)->end(); ++iter)
             {
                 if (iter->first == contextId)
                 {
                     GLuint frameBuffer = iter->second;
                     glCheck(GLEXT_glDeleteFramebuffers(1, &frameBuffer));
 
                     // Erase the entry from the RenderTextureImplFBO's map
-                    (*frameBuffersIter)->erase(iter);
+                    iter = (*frameBuffersIter)->erase(iter);
```

Comment:
I don't think this is correct; we're skipping an element with the increment in the `for` header.
Shouldn't it be:
```cpp
if (...)
{
    ...
    iter = (*frameBuffersIter)->erase(iter);
}
else
{
    ++iter;
}
```

Reply:
You are right, I didn't think this through. Will fix

Reply:
Actually, I think this has always been fine... there's a `break;` right after the `erase` call. I might just revert to the prior implementation

Reply:
Ah, true. Too bad [`erase_if()`](https://en.cppreference.com/w/cpp/container/unordered_map/erase_if) is only part of C++20.

Could you maybe add a comment above the inner for loop, something along the lines "remove first frame buffer which matches context ID"?

Reply:
I'd rather revisit this code and do that in a separate PR. I think there's room for improvement here beyond adding a simple comment.

Reply:
Makes sense, maybe we can also revisit it after the range-based-for / auto iterator PR.

Reply:
Yup, that would be my preference. Let me know if you are happy with the current state of the PR :)